database/database_clouding.py

```diff
@@ -0,0 +1,132 @@
+from google.cloud import spanner
+
+# Instantiates a client
+spanner_client = spanner.Client()
+
+# Your Cloud Spanner instance ID.
+instance = spanner_client.instance('salk-demo')
+
+# Get a Cloud Spanner database by ID.
+database = instance.database("users")
+
+# Dummy data
+def add_vm_to_database(hostname: str): 
+    """
+    Adds a hostname of a Virtual Machine to the database with a default pin and command (None)
+
+    Parameters:
+    hostname (str): The hostname of the VM to be added to the database
+    """
+    with database.batch() as batch:
+        # Checks if there is already a VM with the given hostname
+        with database.snapshot() as snapshot:
+            result = snapshot.execute_sql(
+                "SElECT * FROM Users WHERE HostName = @hostname",
+                params={'hostname': hostname},
+                param_types={'hostname': spanner.param_types.STRING}
+            )
```

Comment:
For integration with the `SpannerDatabase` class, lets store the `table_name` as a variable that is passed into the function.

There is probably a reason that they advise us to pass in parameters in the way above... But, perhaps we could store the query in a variable like so:
```python
query = f"SELECT * FROM {table_name} WHERE HostName = {hostname}"
```

and then

```suggestion
            result = snapshot.execute_sql(query)
```